### PR TITLE
Revert "🤖 Merge PR #49622 [analytics-node] Correct message callback t…

### PR DESCRIPTION
**Pull request by benjaminhoffman**
…ype by @nicklasmoeller"

This reverts commit 826041049fb0f02f5d4f5874f57e78b8b563cc77.

For context, see here:
- Segment introduced a second param here: https://github.com/segmentio/analytics-node/pull/315
- I show that it works according to their documentation in my comment here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49622#issuecomment-1032177010

cc @nicklasmoeller

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by nicklasmoeller** [comment]
Hi @benjaminhoffman. Thanks for tagging me.

As you probably noticed, it landed in https://github.com/segmentio/analytics-node/commit/f5a13857091bed55c02d427de69d94ae3be014db and still hasn't reached a release yet. However, when it does, this definitely makes sense to get merged in.

Your tests in https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49622#issuecomment-1032177010 may have been due to the callback from your message methods (e.g. through `enqueue`) being passed to `flush` as its own callback. That behavior is also changed on the non-released `master` branch - the same commit that I referenced before.

You may be seeing this behavior since the first message in the queue would bind its callback to the flush callback in a good handful of scenarios:
- [on the first message](https://github.com/segmentio/analytics-node/blob/8067d8ba3b2f0fcc791fb896200f61c5e2e5df9b/index.js#L214)
- [when the max items or queue size has been met](https://github.com/segmentio/analytics-node/blob/8067d8ba3b2f0fcc791fb896200f61c5e2e5df9b/index.js#L221)
- [for the periodic flushing that gets set up when a message is enqueued](https://github.com/segmentio/analytics-node/blob/8067d8ba3b2f0fcc791fb896200f61c5e2e5df9b/index.js#L226)

Hence, what you're seeing, should be the callback provided to the `flush` method, through delegation from the `enqueue` method. You should be able to confirm this, by running three `track` (or similar) invocations with callbacks, and you'd only see the first two callbacks if you stick to the default configuration, where the first message gets flushed immediately, and then the second message's callback gets bound to the flush callback, that's bound to the timer.

I acknowledge that it doesn't feel super transparent what's going on, but I hope that the explanation clears things up a bit for you.

I'd say that the changes live up to the current situation of the `master` branch on of `analytics-node`, given that the callbacks are now being invoked with the queue batch data, as of [these lines of code](https://github.com/segmentio/analytics-node/blob/3b9542b4ce06490627d48ee380cdd1b59898cc59/index.js#L265-L270). It's not a tagged release, nor released to npm yet, so I don't know what the practice is with DefinitelyTyped in such situations.

TLDR; This covers unreleased code, but once Segment decides to release their next version after 6.0.0, these changes are 👍 

cc maintainers @fongandrew and @thomasthiebaud

**Comment by benjaminhoffman** [comment]
@nicklasmoeller I agree! ... and came to a similar conclusion _after_ submitting this change. I figured Segment would release soon but I guess not. 🤷 

Would also love if there was a type for the [flush prop](https://github.com/segmentio/analytics-node/blob/master/index.js#L52
), as seen here in [this example](https://github.com/segmentio/analytics-node/issues/309#issuecomment-1032702580).

**Comment by RyanCavanaugh** [comment]
I'm going to approve and the bot is going to ask you to merge, but I'll ask that you hold off on merging until the upstream change is live. If this PR times out you might end up needing to re-open it, unfortunately - the automation 'round these parts isn't set up for that sort of situation.

**Review by RyanCavanaugh — APPROVED** [review]
